src/js/conntest.js

```diff
@@ -40,32 +40,62 @@ function hostConnectivityTest() {
 }
 
 function runConnectivityTest(iceCandidateFilter, config) {
+  var candidates = [];
   var call = new Call(config);
   call.setIceCandidateFilter(iceCandidateFilter);
+
+  // Collect all candidate for validation.
+  call.pc1.onicecandidate = function(event) {
+    if (event.candidate) {
+      candidates.push(Call.parseCandidate(event.candidate.candidate));
```

Comment:
I think the logic is wrong here - if the relay test fails for example, won't you still have gathered reflexive candidates which will end up triggering the warning?

What do you think about filtering and logging here (quite nice to visualize which IPs are being used for each test) and use a separate mechanism to detect that the current test is the reflexive one in hangup?

Maybe something like:
```
if (error && candidates.length && iceCandidateFilter === Call.isReflexive) {
  reportWarning('...')
```